Src/ILGPU.Tests/MemoryBufferOperations.tt

```diff
@@ -0,0 +1,396 @@
+﻿<#@ template debug="false" hostspecific="true" language="C#" #>
+<#@ include file="Generic/ConfigurationBase.tt" #>
+<#@ assembly name="System.Core" #>
+<#@ import namespace="System.IO" #>
+<#@ output extension=".cs" #>
+using ILGPU.Runtime;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using Xunit;
+using Xunit.Abstractions;
+
+<#
+var copyOperations = new string[] { "CopyFrom", "CopyTo" };
+var arrayTypes = new string[] { "Jagged", "Multidimensional" };
+#>
+namespace ILGPU.Tests
+{
+    public abstract class MemoryBufferOperations : TestBase
+    {
+        protected MemoryBufferOperations(ITestOutputHelper output, TestContext testContext)
+            : base(output, testContext)
+        { }
+
+        private static int[] InitialiseArray1D(int x)
+        {
+            var counter = 0;
+            var src = new int[x];
+            for (var i = 0; i < x; i++)
+                src[i] = counter++;
+
+            return src;
+        }
+
+        [SuppressMessage("Performance", "CA1814:Prefer jagged arrays over multidimensional", Justification = "Required for test cases")]
```

Comment:
Let's add some line breaks here like in [BasicJumps.cs](https://github.com/m4rs-mt/ILGPU/blob/master/Src/ILGPU.Tests/BasicJumps.cs#L16).

Reply:
Done